File: AbstractFactoryDesignPattern/GUIComponentCreator_Example/Factory/MacFactory.java
package AbstractFactoryDesignPattern.GUIComponentCreator_Example.Factory;

import AbstractFactoryDesignPattern.GUIComponentCreator_Example.Product.Button;
import AbstractFactoryDesignPattern.GUIComponentCreator_Example.Product.MacButton;
import AbstractFactoryDesignPattern.GUIComponentCreator_Example.Product.MacTextBox;
import AbstractFactoryDesignPattern.GUIComponentCreator_Example.Product.TextBox;

public class MacFactory implements Factory {
	@Override
	public Button createButton() {
		return new MacButton();
	}

	@Override
	public TextBox createTextBox() {
		return new MacTextBox();
	}
}

File: AbstractFactoryDesignPattern/GUIComponentCreator_Example_Python/Factory/MacFactory.py
from AbstractFactoryDesignPattern.GUIComponentCreator_Example_Python.Factory.FactoryInterface import FactoryInterface
from AbstractFactoryDesignPattern.GUIComponentCreator_Example_Python.Product.Button import Button
from AbstractFactoryDesignPattern.GUIComponentCreator_Example_Python.Product.MacButton import MacButton
from AbstractFactoryDesignPattern.GUIComponentCreator_Example_Python.Product.MacTextBox import MacTextBox
from AbstractFactoryDesignPattern.GUIComponentCreator_Example_Python.Product.TextBox import TextBox


class MacFactory(FactoryInterface):

	def createButton(self) -> Button:
		return MacButton()

	def createTextBox(self) -> TextBox:
		return MacTextBox()
